### PR TITLE
Add benchmark instructions page

### LINKED ISSUE #770: Add documentation for the Benchmark suite
This should live under the `Development` tab

### PR DESCRIPTION
**Pull request by szarnyasg**
Fixes #770

### CODE REVIEW

File: dev/benchmark.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+- this is measured
+
+
+
+#### Clone the repo and compile the benchmark runner
+
+```
+git clone https://github.com/duckdb/duckdb
+cd duckdb
+BUILD_BENCHMARK=1 BUILD_TPCH=1 make
```
Comment (carlopi) [inline dev/benchmark.md:27]: Potentially it could be handy mentioning LTO='thin' here. Should work on clang or gcc.

File: dev/benchmark.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+#### Regex
+You can also use a regex to specify which benchmarks to run. Be careful of shell expansion of certain regex characters (e.g. `*` will likely be expanded by your shell, hence this requires proper quoting or escaping).
+
+`build/release/benchmark/benchmark_runner "benchmark/micro/nulls/.*" `
+
+#### Run all benchmarks
+Not specifying any argument will run all benchmarks.
+
+`build/release/benchmark/benchmark_runner`
```
Comment (carlopi) [inline dev/benchmark.md:67]: Will this actually work?
I guess we don't care, but I would expect this to fail somewhere.

I will also point out that CI makes sure some benchmarks (TPCH / TPCDS / IMDB + a few others) are run on any PRs to check for regressions.
And I would explain what / how to reproduce the CI runs (pointing to csv files in .github/regression).

**Review by carlopi — COMMENTED** [review]
Nice addition.

File: dev/benchmark.md (hunk 1 above)
Comment (Tmonster) [reply dev/benchmark.md:16]: maybe `BUILD_TPCDS` as well?

**Review by Tmonster — APPROVED** [review]
lgtm!

**Pull request merged**